app/controllers/oauth/applications_controller.rb

```diff
@@ -109,12 +121,49 @@ def set_user
     end
 
     def app_params
-      params.require(:doorkeeper_application).permit(
-        :name, :redirect_uri, :scopes, :email_subject_prefix, :lead_application_source,
-        :email_from_address, :confidential,
-        :can_access_private_user_data, :can_find_or_create_accounts, :can_message_users,
-        :can_skip_oauth_screen,
-      )
+      if @user.is_administrator?
+        params.require(:doorkeeper_application).permit(
+          :name, :redirect_uri, :scopes, :email_subject_prefix, :lead_application_source,
+          :email_from_address, :confidential,
+          :can_access_private_user_data, :can_find_or_create_accounts, :can_message_users,
+          :can_skip_oauth_screen, :member_ids
+        )
+      elsif @application.owner.member_ids.include?(@user.id)
+        params.require(:doorkeeper_application).permit(
+          :redirect_uri
+        )
+      end
+    end
+
+    def handle_member_ids(member_ids)
+      if valid_member_ids(member_ids)
+        if !member_ids.nil?
+          str_array = member_ids.split(" ")
+          mi_array = str_array.map(&:to_i)
```

Comment:
I know above I recommended a different way to approach this, but if this were to stay the same I'd recommend some different naming conventions:

1. Make clear that the input to the method is a string and not an array of member IDs: `if valid_member_ids(member_ids)` --> `if valid_member_ids(member_ids_string)`
2. Avoid naming things "_array", just make them plural which implies an array: `mi_array` -> `member_ids` 

And as shown above, chaining short commands is a typical Ruby thing (Ruby values conciseness); you can eliminate the `str_array` here with `member_ids_string.split.map(&:to_i)`

Reply:
Got it